### PR TITLE
add in ability to lock simulator version

### PR DESCRIPTION
**Pull request by cindydeng1998**
**Task name**: AzureIoTEdgeV2

**Description**: 
- add in logic to set iotedgehubdev version via environment variable

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) N

**Attached related issue:** (Y/N) N

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/AzureIoTEdgeV2/util.ts [hunk 1]
```diff
@@ -80,25 +80,43 @@ export default class Util {
...
     if (tl.osType() === Constants.osTypeLinux) {
       cmds = [
         { path: `sudo`, arg: `apt-get update`, execOption: Constants.execSyncSilentOption },
         { path: `sudo`, arg: `-H pip3 install wheel --upgrade`, execOption: Constants.execSyncSilentOption },
         { path: `sudo`, arg: `pip3 install importlib-metadata==2.1.1`, execOption: Constants.execSyncSilentOption },
         { path: `sudo`, arg: `apt-get install -y python3-setuptools`, execOption: Constants.execSyncSilentOption },
-        { path: `sudo`, arg: `pip3 install ${Constants.iotedgedev}~=${version}`, execOption: Constants.execSyncSilentOption },
       ]
+
+      if (tl.getVariable(Constants.iotedgedevLockVersionKey)) {
```
Comment (mhshami01) [inline Tasks/AzureIoTEdgeV2/util.ts:94]: consider using local variables

**Review by mhshami01 — APPROVED** [review]

File: Tasks/AzureIoTEdgeV2/util.ts (hunk 1 above)
Comment (marianan) [inline Tasks/AzureIoTEdgeV2/util.ts:94]: >tl.getVariable(Constants.iotedgedevLockVersionKey [](http://example.com/codeflow?start=10&length=49)

the code reads this variable multiple times; unless we expect it to change, maybe cache it in a local variable?

File: Tasks/AzureIoTEdgeV2/util.ts [hunk 2]
```diff
@@ -80,25 +80,43 @@ export default class Util {
     }
 
     let cmds: Cmd[] = [];
-    let version = Constants.iotedgedevDefaultVersion;
-    if (tl.getVariable(Constants.iotedgedevLockVersionKey)) {
-      version = tl.getVariable(Constants.iotedgedevLockVersionKey);
-    }
-    tl.debug(`The specified iotedgedev version is: ${version}`);
+    let edgeDevVersion = Constants.iotedgedevDefaultVersion;
+    let simulatorVersion : string = null;
```
Comment (marianan) [inline Tasks/AzureIoTEdgeV2/util.ts:84]: >simulatorVersion [](http://example.com/codeflow?start=8&length=16)

Add a comment that the simulator is expected to be installed by iotedgedev

**Review by marianan — APPROVED** [review]
:shipit:

**Review by mhshami01 — APPROVED** [review]

**Pull request merged**